cmake/Modules/FindRTIConnextDDS.cmake

```diff
@@ -481,31 +477,35 @@ if(POLICY CMP0074)
     cmake_policy(SET CMP0074 NEW)
 endif()
 
+# Starting with CMake 3.3, the support for the IN_LIST operator in the if block
+# was added. The OLD behaviour is to ignore this operator, so we set the policy
+# to NEW.
+if(POLICY CMP0057)
+    cmake_policy(SET CMP0057 NEW)
+endif()
+
 #####################################################################
 # Global Variables                                                  #
 #####################################################################
 
-option(CONNEXTDDS_IMPORTED_TARGETS_DEBUG
-    "Force the linker to use the Connext debug libraries"
-    OFF
-)
-option(CONNEXT_USE_GLOBAL_BUILD_TYPE
-    "Enforce the Connext libraries build type speficied by the global\
- CMAKE_BUILD_TYPE variable"
-    OFF
-)
-set(CONNEXTDDS_LOG_LEVEL
-    "STATUS"
-    CACHE
-    STRING
-    "Set the logging level for the script"
+set(CONNEXT_LIBS_BUILD_TYPE_LIST "Release" "Debug" "Auto")
+set(CONNEXT_LIBS_BUILD_TYPE "Auto" CACHE STRING
+    "Connext imported target libraries build type to use"
 )
 
-if(CONNEXT_USE_GLOBAL_BUILD_TYPE AND CONNEXTDDS_IMPORTED_TARGETS_DEBUG)
-    message(WARNING
-        "WARNING When using the `CONNEXT_USE_GLOBAL_BUILD_TYPE`, the"
-        " `CONNEXTDDS_IMPORTED_TARGETS_DEBUG` variable will not have any"
-        " effect. Only the global CMAKE_BUILD_TYPE will be taken into account."
+if(CONNEXTDDS_IMPORTED_TARGETS_DEBUG)
+    message(DEPRECATION
```

Comment:
Do we have a cmake minimum version? The `DEPRECATION` message was introduced in 3.17. If you use an older version you will get a syntax error

Reply:
I found the `DEPRECATION` message in 3.0: https://cmake.org/cmake/help/v3.0/command/message.html?highlight=message.

Reply:
My fault. The `DEPRECATION` property is what was added in 3.17.